linera-core/src/client/mod.rs

```diff
@@ -1740,7 +1737,8 @@ where
     /// Attempts to execute the block locally. If any incoming message execution fails, that
     /// message is rejected and execution is retried, until the block accepts only messages
     /// that succeed.
-    #[instrument(level = "trace", skip(block))]
+    // TODO: Measure how failing messages affect the execution times.
```

Comment:
Let's create an issue for that and link to it.

Also, I think we want to change this in the long run regardless: Even without measuring we know it's `O(N²)`, so at the very least we'd like to take a snapshot of the execution state after the first N message bundles before trying the N+1st, so that if that fails, we don't have to re-execute the first N.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/2745/commits/b5c6ebf561827dffa2aa7d036a59f80549f54b87.